### DISCUSSION TITLE
Snapshot, zones, best image

### DISCUSSION THREAD

**Discussion by Raphael909** [blakeblackshear/frigate#1983] [Ask A Question]
Hello folks, 
I was trying to wrap my head around concept of snapshot, combined with required zones and best image score.
To my understanding, if I have zone say zone_0, triggered only by object - person, configured under camera level snapshot configuration (zone_0 is in required_zone also), then best image will be stored based on the best score within zone_0 .. In other words, person image will not be store outside of zone_0. Is this correct?
If it is,  then for some reason, Frigate is storing best image of the person outside of zone_0 ( close to doorbell camera). 
If not, then would it be possible to achieve the goal of only capturing image of a person within zone_0.

Thank you !

**Comment by blakeblackshear** [comment]
Keep in mind that frigate detects objects and then tracks them over many frames.
- `snapshot` - the "best" image of the tracked object, regardless of where it was. it tries to balance the size, score, and age of the frames to find the best. it's not always the highest scoring
- `required_zones` - the tracked object must have entered the zone for at least one frame in order to save it's snapshot. It has nothing to do with determining where the object was located in the snapshot.
- `best image` - I'm not totally sure what you are referring to, but this is probably just another word for snapshot as defined above.

The snapshot may not be from when the object was in the zone. It only matters if that object entered a zone in any frames where it was tracked.

**Reply by Raphael909** [reply]
> Keep in mind that frigate detects objects and then tracks them over many frames.
> 
> * `snapshot` - the "best" image of the tracked object, regardless of where it was. it tries to balance the size, score, and age of the frames to find the best. it's not always the highest scoring
> * `required_zones` - the tracked object must have entered the zone for at least one frame in order to save it's snapshot. It has nothing to do with determining where the object was located in the snapshot.
> * `best image` - I'm not totally sure what you are referring to, but this is probably just another word for snapshot as defined above.
> 
> The snapshot may not be from when the object was in the zone. It only matters if that object entered a zone in any frames where it was tracked.

Looks like my understanding is bit off .

The problem I am trying to solve is, when frigate detects a person in front of my door steps ( zone_0), it triggers a HA notification (using recommend blueprint). Notification has person image attached which keeps changing over time ( for that ID). As the person move closer to front door (and close to doorbell camera), I loose person face. Frigate saves this image to disk but it is not very useful by then.
Thinking of this I created zone_0, few steps away from door so that I can capture person in it's entirety but looks like this is not the right way to solve the problem.

Any thoughts on how I can go about solving this?

**Reply by blakeblackshear** [reply]
This might help:
![image](https://user-images.githubusercontent.com/569905/136713181-07c24cac-f70c-407f-a7f7-647788775864.png)
